tsdb/querier.go

```diff
@@ -365,29 +365,16 @@ func inversePostingsForMatcher(ctx context.Context, ix IndexReader, m *labels.Ma
 		return ix.Postings(ctx, m.Name, m.Value)
 	}
 
-	vals, err := ix.LabelValues(ctx, m.Name)
-	if err != nil {
-		return nil, err
-	}
-
-	res := vals[:0]
-	// If the match before inversion was !="" or !~"", we just want all the values.
+	// If inverse matching the empty string, we just want all the values.
```

Comment:
To me this seems harder to understand than the original comment which spelled out the cases we are trying to handle:

	// If the match before inversion was !="" or !~"", we just want all the values.

Is there a version better than both?

Reply:
> // If the match before inversion was !="" or !~"", we just want all the values.
 
 The old comment is misleading though isn't it, since the cases being inverted are `=~""` and `=""`. That's why I thought my comment was more correct.

Reply:
Yes, that is a lot better. I didn't detect that the previous comment was opposite to yours.